gamemodes/terrortown/entities/entities/ttt_base_placeable.lua

```diff
@@ -306,12 +306,19 @@ if SERVER then
     -- @return boolean Returns true on success
     -- @realm server
     function ENT:ThrowEntity(ply, rotationalOffset)
+        local posThrow = ply:GetShootPos() - Vector(0, 0, 15)
+        local vecAim = ply:GetAimVector()
+
+        if not ply:HasDropSpace(nil, posThrow, vecAim) then
```

Comment:
For whatever reasons this function needs a weapon. The entity to throw should also be included here or not?

Reply:
no, as the weapon drop and the entity throw are really different imho

Reply:
But doesnt that check just include the entity in the trace as a filter, so it doesnt count as blocking entity?
Should be the same for weapon and placeable.

Reply:
I don't have the reference for the weapon in this case. I could loop in through, but that is not needed. Probably it isn't needed in the filter anyway, as it wasn't part of the filter before my change, but I added it where I had it because there might be some weird edge cases with really big weapons

Reply:
isnt it just self?

Reply:
no, self is the entity that is about to be spawned, not the weapon the player is holding